include/rocksdb/iterator.h

```diff
@@ -92,7 +92,9 @@ class Iterator : public Cleanable {
   // If supported, renew the iterator to represent the latest state. The
   // iterator will be invalidated after the call. Not supported if
   // ReadOptions.snapshot is given when creating the iterator.
-  virtual Status Refresh() {
+  virtual Status Refresh() { return Refresh(nullptr); }
+
+  virtual Status Refresh(const class Snapshot*) {
```

Comment:
> It might be worth clarifying if the parameter needs to be the same snapshot used in read options to create the original iterator. If so, maybe there is no need to pass in snapshot since we can access it from ArenaWrappedDBIter.read_options_?

You are right, it is no need to pass a same `snapshot` object, snapshot in ArenaWrappedDBIter.read_options_ should be updated as the passing in `snapshot`.

Reply:
IMO - does not need to be the same, but we should anyways require the user provide snapshot, regardless of whether there was one in `ReadOptions`. That way we can guarantee the target snapshot still exists so we don't accidentally refresh to a SuperVersion on which that snapshot is unavailable.

I am glad that the no-argument `Refresh()` refreshes to latest state rather than the `ReadOptions::snapshot`. We need to make it clear from the API doc though.